### PR TITLE
fix: use ADBC execute_update for DDL/DML wire OK responses

### LINKED ISSUE #97: MySQL Wire Protocol: DDL/DML operations succeed on backend but client receives ERROR 2008 (HY000) and intermittent ERROR 2013 (HY000)
Environment

QueryFlux: latest container image (ghcr.io/lakeops-org/queryflux:latest)
Backend: Apache Doris
Engine: ADBC
Driver: /usr/local/lib/adbc/mysql_linux_amd64_v0.4.0/libadbc_driver_mysql.so

QueryFlux persistence: PostgreSQL
Frontend: MySQL Wire (3306)

Cluster Configuration
JSON{  "engineKey": "adbc",  "enabled": true,  "config": {    "driver": "/usr/local/lib/adbc/mysql_linux_amd64_v0.4.0/libadbc_driver_mysql.so",    "uri": "mysql://admin:<password>@<doris-fe>:9030/mysql",    "poolSize": 4  }}

Cluster state:
JSON{  "group_name": "Doris",  "cluster_name": "DorisADBC",  "engine_type": "Adbc",  "is_healthy": true,  "enabled": true}Show more lines

Summary
I was able to successfully configure QueryFlux to route MySQL clients to Apache Doris through ADBC. Query execution works and data is persisted correctly.
However, DDL and DML statements return errors to the MySQL client even though they are successfully executed on Doris.
Additionally, some metadata commands cause the client connection to drop and reconnect automatically before returning results.

Reproduction

Create a table:

CREATE TABLE benchmark.TESTE_MEDIUM (    ID MEDIUMINT,    NOME VARCHAR(100))DUPLICATE KEY(ID)DISTRIBUTED BY HASH(ID) BUCKETS 1PROPERTIES (    "replication_num" = "1");

Client receives:
ERROR 2008 (HY000): MySQL client ran out of memory
However:
SHOW TABLES FROM benchmark;
shows:
TESTE_MEDIUM
meaning the table was actually created.

Insert data:
INSERT INTO TESTE_MEDIUMVALUES (1, 'TESTE');

Client receives:

ERROR 2008 (HY000): MySQL client ran out of memoryShow more lines
But:

SELECT *FROM TESTE_MEDIUM;Show more lines
returns:

Text+------+-------+| ID   | NOME  |+------+-------+|    1 | TESTE |+------+-------+Show more lines
showing that the INSERT was successfully executed.

Drop table:
SQLDROP TABLE benchmark.TESTE;Show more lines

Client receives:

Yet:

confirms that the table was actually dropped.

Additional Issue
Metadata commands sometimes disconnect the client:

returns:

ERROR 2013 (HY000): Lost connection to MySQL server during queryNo connection. Trying to reconnect...

After reconnection, the result set is returned correctly.

Example:
Plain Text+---------------------+| Tables_in_benchmark |+---------------------+| IFTD_AGENCIA        || TESTE_INT           || TESTE_MEDIUM        || vendas              |+---------------------+

What Works
✅ Connection through MySQL Wire frontend
✅ Routing to Doris via ADBC
✅ SELECT statements
✅ SHOW TABLES
✅ DESCRIBE
✅ CREATE TABLE (executed on backend)
✅ INSERT (executed on backend)
✅ DROP TABLE (executed on backend)
✅ Persistence through PostgreSQL

Expected Behavior
Successful DDL and DML operations should return standard MySQL OK packets instead of:
ERROR 2008 (HY000): MySQL client ran out of memory
Metadata operations should not force client reconnections.

Clients Tested
Issue reproduced with:

MySQL CLI

Issue also reproduced when connecting from outside the Kubernetes/OpenShift cluster through a NodePort service.

Observation
This appears to be a MySQL Wire protocol response issue rather than an execution issue. The backend operation is completed successfully, but the client receives an error or connection reset while processing the response.

### PR DESCRIPTION
**Pull request by amitgilad3**
## Summary

- **Root cause:** ADBC adapter called `stmt.execute()` for all SQL statements, including DDL/DML. These return empty Arrow streams, which dispatch synthesized as `Schema::empty()`. The MySQL wire frontend encoded column-count `0` as `0x00` — indistinguishable from the OK marker — causing `ERROR 2008 (HY000)` and intermittent `ERROR 2013`.
- **Fix:** Route DDL/DML statements through ADBC's `execute_update()` API (the standard path for non-result-set statements), propagate `affected_rows` through `SyncExecution` → `QueryStats`, and add `ResponseMode` defense to MySQL and Postgres wire sinks so empty/missing schemas always emit OK/CommandComplete instead of broken result-set framing.
- **Scope:** All wire protocols (MySQL, Postgres). Flight SQL, Trino HTTP, and Snowflake sinks verified compatible. Other adapters (DuckDB, StarRocks, Athena, Trino) unaffected — they set `affected_rows: None`.

## Changes

| File | What |
|------|------|
| `adbc/mod.rs` | SQL classification heuristic (`is_read_like_sql`); SELECT/SHOW/DESCRIBE/EXPLAIN/WITH → `execute()`, else → `execute_update()` returning `affected_rows` |
| `lib.rs` (engine-adapters) | `SyncExecution.affected_rows: Option<u64>` |
| `query.rs` (core) | `QueryStats.affected_rows: Option<u64>` |
| `dispatch.rs` | Stop synthesizing `on_schema(Schema::empty())`; pass `affected_rows` into `QueryStats` |
| `mysql_wire/mod.rs` | `ResponseMode { None, ResultSet, Ok }` — empty/no schema → `build_ok(affected_rows)`; non-empty schema → `build_eof()` |
| `postgres_wire/mod.rs` | Skip `RowDescription` for empty schema; `CommandComplete` uses `OK {n}` for DDL/DML, `SELECT {n}` for result sets |
| `mysql-wire.md` | Doc note on DDL/DML OK packet behavior |
| Other adapters | `affected_rows: None` added to `SyncExecution` constructors (no behavior change) |

## Test plan

- [x] ADBC `is_read_like_sql` unit tests (SELECT, WITH, SHOW, DESCRIBE, EXPLAIN vs CREATE, INSERT, DROP, etc.)
- [x] `strip_leading_block_comments` edge cases
- [x] MySQL sink: no-schema complete → OK packet (with affected_rows), empty schema → OK mode, non-empty schema → ResultSet + EOF
- [x] Postgres sink: no-schema complete → `OK` CommandComplete, empty schema → no RowDescription, non-empty → `SELECT` tag
- [x] `cargo check` + `cargo clippy -D warnings` clean
- [x] All existing tests pass (81 tests)

Fixes #97

Made with [Cursor](https://cursor.com)

## Summary by CodeRabbit

* **New Features**
  * Added affected-row reporting for DDL/DML statements.
  * Added SQL statement classification and execution hints.
  * Added configurable Athena wait limits, DuckDB pooling, result buffering, and cancellation.
  * Added authenticated query passthrough and backend query tracking across supported adapters.

* **Bug Fixes**
  * Improved MySQL and PostgreSQL responses for statements without result sets.
  * Preserved schemas for empty query results and improved streaming.
  * Improved timeout, cancellation, authentication, and disconnect handling.

* **Documentation**
  * Documented DDL/DML affected-row behavior for MySQL clients.

### CODE REVIEW

File: crates/queryflux-engine-adapters/src/adbc/mod.rs [hunk 1]
```diff
@@ -478,6 +478,35 @@ fn collect_batches(
...
+        if let Some(rest) = s.strip_prefix("/*") {
+            match rest.find("*/") {
+                Some(end) => s = &rest[end + 2..],
+                None => return s,
+            }
+        } else {
+            return s;
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (amitgilad3) [reply crates/queryflux-engine-adapters/src/adbc/mod.rs]: Fixed in 64f3c2c:

- `strip_leading_sql_comments` now strips leading `--` line comments as well as `/* … */` block comments (including repeated / mixed forms).
- Docstring updated to describe the actual heuristic (no longer claims to mirror `ReadOnlyGuard`).
- Added tests covering `--` alone, repeated `--`, and `--` followed by block comments before SELECT.

File: crates/queryflux-frontend/src/mysql_wire/mod.rs [hunk 2]
```diff
@@ -622,6 +640,13 @@ impl MysqlResultSink {
 #[async_trait]
 impl ResultSink for MysqlResultSink {
     async fn on_schema(&mut self, schema: &Schema) -> Result<()> {
+        if schema.fields().is_empty() {
+            self.mode = ResponseMode::Ok;
+            return Ok(());
+        }
+
+        self.mode = ResponseMode::ResultSet;
+
```
[the remark this answers is not in the document]
Comment (amitgilad3) [reply crates/queryflux-frontend/src/mysql_wire/mod.rs:706]: Fixed in 64f3c2c:

- MySQL `on_batch` is a no-op unless `ResponseMode::ResultSet`.
- Empty native `columns` sets `ResponseMode::Ok` and skips zero column-count / row packets.
- Postgres `on_batch` is a no-op unless `schema_sent` (no DataRow without RowDescription).
- Added tests for empty-schema + batch and empty native columns.

File: crates/queryflux-engine-adapters/src/duckdb/http.rs [hunk 3]
```diff
@@ -119,42 +124,80 @@ pub struct DuckDbHttpAdapter {
...
+        return DataType::Utf8;
+    }
+    let base = ty.split('(').next().unwrap_or(ty).trim().to_uppercase();
+    match base.as_str() {
+        "BOOLEAN" => DataType::Boolean,
+        "TINYINT" | "SMALLINT" | "INTEGER" | "BIGINT" | "HUGEINT" | "UTINYINT" | "USMALLINT"
+        | "UINTEGER" | "UBIGINT" | "UHUGEINT" => DataType::Int64,
+        "FLOAT" | "DOUBLE" | "DECIMAL" => DataType::Float64,
+        _ => DataType::Utf8,
     }
```
[the remark this answers is not in the document]
Comment (amitgilad3) [reply crates/queryflux-engine-adapters/src/duckdb/http.rs:208]: Fixed in 51201a1: `duckdb_type_to_arrow` now maps HUGEINT/UBIGINT/UHUGEINT to `Utf8` instead of `Int64`, matching what DuckDB's own serializer actually emits on the wire. Added a regression test with a UBIGINT value beyond i64::MAX confirming it round-trips as text instead of becoming NULL.

File: crates/queryflux-frontend/src/postgres_wire/mod.rs [hunk 4]
```diff
@@ -430,18 +430,104 @@ async fn handle_simple_query(
...
+                verb: "OK".to_string(),
+                has_row_count: false,
+            },
+            other => Self {
+                verb: other.to_string(),
+                has_row_count: false,
+            },
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (amitgilad3) [reply crates/queryflux-frontend/src/postgres_wire/mod.rs:516]: Fixed in 51201a1, using the suggested fix (trim_end_matches on non-alphanumeric chars before uppercasing). Added COMMIT;/VACUUM;/BEGIN; to the classification test table.

File: crates/queryflux-core/src/polyglot_pool.rs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            let task = err.0;
+            std::thread::Builder::new()
+                .stack_size(POLYGLOT_STACK_SIZE)
+                .spawn(task)
+                .ok()?
+                .join()
+                .ok()?;
+            done_rx.recv().ok()
+        }
+    }
```
[the remark this answers is not in the document]
Comment (amitgilad3) [reply crates/queryflux-core/src/polyglot_pool.rs:72]: Fixed in 51201a1: added `SqlParseCache::ensure_parsed`/`is_read_like_async`/`statements_async`, which run the parse (queue + pool wait) via `tokio::task::spawn_blocking` and await it instead of blocking the caller's worker thread. Updated dispatch's `execute_stream` and all three built-in guards (plus the no-shared-cache fallback path in `guard_statements`) to use the async accessors. Required adding `tokio` as a direct dependency of `queryflux-core`.
